src/transformers/adapters/peft_mixin.py

```diff
@@ -0,0 +1,333 @@
+                for the same reasons stated above.
+            offload_dir (`str`, `optional`):
+                The directory to use for offloading the model. Used only in the case the model is offloaded into the
+                CPU.
+            offload_index (`int`, `optional`):
+                `offload_index` argument to be passed to `accelerate.dispatch_model` method.
+        """
+        check_peft_version(min_version="0.4.0")
+
+        adapter_name = adapter_name or "default"
+
+        from peft import PeftConfig, inject_adapter_in_model, load_peft_weights
+        from peft.utils import set_peft_model_state_dict
+
+        if not self._hf_peft_config_loaded:
+            self._hf_peft_config_loaded = True
+        elif adapter_name in self.peft_config:
+            raise ValueError(f"Adapter with name {adapter_name} already exists. Please use a different name.")
+
+        adapter_config_file = find_adapter_config_file(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+            commit_hash=commit_hash,
+        )
+
+        if adapter_config_file is None:
+            raise ValueError(
+                f"adapter model file not found in {peft_model_id}. Make sure you are passing the correct path to the "
+                "adapter model."
+            )
+
+        loaded_peft_config = PeftConfig.from_pretrained(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+            commit_hash=commit_hash,
+        )
+
+        # Replace the adapter with the loaded adapter
+        inject_adapter_in_model(loaded_peft_config, self, adapter_name)
+
+        adapter_state_dict = load_peft_weights(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+        )
+
+        # We need to pre-process the state dict to remove unneeded prefixes - for backward compatibility
+        processed_adapter_state_dict = {}
+        for key, value in adapter_state_dict.items():
+            if "base_model.model" in key:
+                new_key = key.replace("base_model.model.", "")
+            else:
+                new_key = key
+            processed_adapter_state_dict[new_key] = value
+
+        # Load state dict
+        incompatible_keys = set_peft_model_state_dict(self, processed_adapter_state_dict, adapter_name)
+
+        if incompatible_keys is not None:
+            # check only for unexpected keys
+            if hasattr(incompatible_keys, "unexpected_keys") and len(incompatible_keys.unexpected_keys) > 0:
+                logger.warning(
+                    f"Loading adapter weights from {peft_model_id} led to unexpected keys not found in the model: "
+                    f" {incompatible_keys.unexpected_keys}. "
+                )
+
+        # @pacman100 why this was needed?
```

Comment:
To be addressed?

Reply:
Actually it is addressed, I have discussed with @pacman100 offline, and this is needed to correctly dispatch the model into CPU /Disk in case the base model is offloaded